src/lib/utils/ct_utils.h

```diff
@@ -64,15 +67,89 @@ constexpr inline void unpoison(const T* p, size_t n) {
    BOTAN_UNUSED(p, n);
 }
 
+//
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+//
+// Convenience overloads for poisoning/unpoisoning various types and ranges
+//
+
+template <std::integral T>
+constexpr void poison(T& p) {
+   poison(&p, sizeof(T));
+}
+
+template <std::integral T>
+constexpr void unpoison(T& p) {
+   unpoison(&p, sizeof(T));
+}
+
+template <ranges::spanable_range R>
+constexpr void poison(R&& r) {
```

Comment:
Why not just take a `span` as the argument?

Reply:
That would mean to template on the span-ed type which does not work with implicit conversion. Like:

```C++
template <typename T>
constexpr void poison(std::span<T> s) {
   /* ... */
}
```

You can't just call that with a vector (say `std::vector<uint32_t> x`) like: `poison(x)`, because it won't deduce the `T` as the `std::ranges::range_value_t<decltype(x)>`.

The `spanable_range` concept restricts to every type for which `std::span{x}` would compile, thus making the conversion to span explicit.